### DISCUSSION TITLE
How to set "Select Linear Text" a shortcut through plugin? How to cancel default shortcuts?

### DISCUSSION THREAD

**Discussion by hormenks** [xournalpp/xournalpp#5588] [Q&A] [answered]
I have successfully set the "b" key as the shortcut for "Pen" function according to the plugin's guide. 
How can I set "Select Linear Text" a shortcut, such as "t" key, in the same way? I am not able to find the right ACTION name of "Select Linear Text".

```
function initUi()
  app.registerUi({["menu"] = "Pen", ["callback"] = "click_pen", ["accelerator"] = "b"});
end

function click_pen()
  app.uiAction({["action"]="ACTION_TOOL_PEN"})
end
```

And thanks for xournal++, it's a very useful software.

**Comment by rolandlo** [comment, marked as answer]
I think `ACTION_TOOL_SELECT_PDF_TEXT_LINEAR` should do the job. See https://github.com/xournalpp/xournalpp/blob/master/src/core/plugin/ActionBackwardCompatibilityLayer.cpp for a list of all actions.

**Reply by hormenks** [reply]
Thank you, it works!!! The present help document seems not to contain this link, I think it's better to add this.

And I wonder if shortcuts can be set as "[" or  "]" or ";" or so on. I have tried, but didn't succeed.
Is the plugin way only suitable for "a-z" key?

**Reply by rolandlo** [reply]
The plugin way is suitable for everything that [`gtk_accelerator_parse`](https://docs.gtk.org/gtk4/func.accelerator_parse.html) can parse. I haven't tried "[" and ";" yet, but [the code](https://gitlab.gnome.org/GNOME/gtk/blob/90d84a2af8b367bd5a5312b3fa3b67563462c0ef/gdk/keynames.txt#L61-L61) in combination with [this source](https://wumbo.net/symbols/left-square-bracket/) suggests you should use "semicolon" instead of ";" and "bracketleft" for "[".

**Reply by hormenks** [reply]
Thanks for your answer! The code like "bracketleft" works successfully. 
I haven't learned about gtk_accelerator_parse, and I am not good at coding too. Your patience helps me a lot!

**Reply by hormenks** [reply]
Hi, I wonder if I can cancel the default shortcut "h" (it has been set as "left" function)?
I tried to set "h" as "hand tool" through plugin, but I failed.

**Reply by rolandlo** [reply]
I don't think you can. The focus widget comes first in the line of handling shortcuts.

**Reply by hormenks** [reply]
Thanks for response. I will set another key.

**Comment by hormenks** [comment]
I find the key number of "H" is different from "h" in gtk_accelerator_parse. So I set the "H" key for hand tool and succeed to avoid the conflict with default "h" shortcut. If anyone wants to cancel the default shortcut but haven't found the way, uppercase key may help.
